report_thesis/src/sections/background/ensemble_learning_models/xgboost.tex

```diff
@@ -1 +1,27 @@
-\subsubsection{XGBoost}
\ No newline at end of file
+\subsubsection{XGBoost}
+We now describe XGBoost based on \citet{ChenGuestrin2016}.
+
+\gls{xgboost} is an advanced implementation of gradient boosting designed for speed and performance. 
+It builds upon traditional boosting techniques by introducing several algorithmic enhancements to improve execution speed and model performance.
+
+The core mechanism of \gls{xgboost} can be summarized in the following way:
+\begin{enumerate}
+\item The algorithms first step involves sorting the feature values and proposing candidate split points based on the distribution of these values.
+\item It then maps these features into buckets based on the candidate splitting points.
```

Comment:
Be careful here because it seems like you mix the approximate and exact greedy algorithms. They are different.

The exact greedy algorithm sorts by feature values and visits the data in sorted order to get gradient statistics for the split evaluation, whereas the approximate algorithm proposes candidate splits by percentiles of the feature distribution, then maps the continuous features to buckets split by the candidate points, aggregates the statistics, and then does the evaluation.

So it seems your step 1 and 2 are mixing up these 2 algorithms?